agent/agent/src/main/java/com/microsoft/applicationinsights/agent/Agent.java

```diff
@@ -40,7 +40,11 @@ public static void premain(String agentArgs, Instrumentation inst) {
     if (alreadyLoaded) {
       return;
     }
-    StartupProfiler.start();
+
+    // this is used to support -Dapplicationinsights.debug.startupProfiling=true
+    if (Boolean.parseBoolean(System.getProperty("applicationinsights.debug.startupProfiling"))) {
```

Comment:
```suggestion
    if (Boolean.getBoolean("applicationinsights.debug.startupProfiling")) {
```

Reply:
getBoolean is calling parseBoolean... i don't see why this is better. System.getProperty is needed though.

Reply:
https://docs.oracle.com/javase/7/docs/api/java/lang/Boolean.html#getBoolean(java.lang.String)